OneSignalSDK/onesignal/in-app-messages/src/main/java/com/onesignal/inAppMessages/internal/backend/impl/InAppBackendService.kt

```diff
@@ -209,4 +198,82 @@ internal class InAppBackendService(
     ) {
         Logging.error("Encountered a $statusCode error while attempting in-app message $requestType request: $response")
     }
+
+    private suspend fun attemptFetchWithRetries(
+        baseUrl: String,
+        offset: Long,
+        sessionTime: Long,
+    ): List<InAppMessage>? {
+        var attempts = 1
+        var delayTime = 1 // Start with a 1-second delay for exponential backoff
+
+        while (true) {
+            val headerValues = OptionalHeaderValues()
+            if (attempts > 1) {
+                headerValues.retryCount = attempts - 1
+            }
+            headerValues.offset = offset
+            headerValues.secondsSinceAppOpen = sessionTime
+            val response = _httpClient.get(baseUrl, headerValues)
+
+            if (response.isSuccess) {
+                val jsonResponse = response.payload?.let { JSONObject(it) }
+                return jsonResponse?.let { hydrateInAppMessages(it) }
+            } else if (response.statusCode == 425) { // 425 Too Early
+                val retryLimit = response.retryLimit ?: 3
```

Comment:
A retry value should not be inline. Either make a retry function that does this logic and/or make a safe default constant.